src/components/GeoChart.tsx

```diff
@@ -0,0 +1,133 @@
+// (C) 2007-2019 GoodData Corporation
+import * as React from "react";
+import { VisualizationObject, VisualizationInput } from "@gooddata/typings";
+import omit = require("lodash/omit");
+import noop = require("lodash/noop");
+import get = require("lodash/get");
+
+import { Subtract } from "../typings/subtract";
+import { GeoChart as CoreGeoChart } from "./core/GeoChart";
+import { dataSourceProvider } from "./afm/DataSourceProvider";
+import { ICommonChartProps } from "./core/base/BaseChart";
+import { convertBucketsToAFM } from "../helpers/conversion";
+import { getGeoChartDimensions } from "../helpers/dimensions";
+import { getResultSpec } from "../helpers/resultSpec";
+
+import { COLOR, LOCATION, SEGMENT, SIZE } from "../constants/bucketNames";
+import { IChartConfig } from "..";
+// import { hasDuplicateIdentifiers } from "../helpers/errorHandlers";
+export interface IGeoDataIndex {
+    location?: number;
+    size?: number;
+    color?: number;
+    segmentBy?: number;
+    tooltipText?: number;
+}
+
+export interface ICoordinates {
+    longtitude: number;
+    lattitude: number;
+}
+
+export interface IGeoChartBucketProps {
+    location?: VisualizationInput.IAttribute;
+    size?: VisualizationInput.AttributeOrMeasure;
+    color?: VisualizationInput.AttributeOrMeasure;
+    segmentBy?: VisualizationInput.IAttribute;
+    filters?: VisualizationInput.IFilter[];
+    sortBy?: VisualizationInput.ISort[];
+}
+
+export interface IGeoChartProps extends ICommonChartProps, IGeoChartBucketProps {
+    projectId: string;
+    exportTitle?: string;
+}
+
+export const getBuckets = (props: IGeoChartProps): VisualizationObject.IBucket[] => {
+    const { color, location, segmentBy, size, config } = props;
+    const bucket = [
+        {
+            localIdentifier: SIZE,
+            items: [size] || [],
```

Comment:
could you please double check this?
it could be `[undefined]` and the later expression `|| []` may be redundant